src/log_surgeon/finite_automata/RegexNFA.hpp

```diff
@@ -310,29 +311,44 @@ void RegexNFAState<state_type>::add_interval(Interval interval, RegexNFAState* d
 template <RegexNFAStateType state_type>
 auto RegexNFAState<state_type>::serialize(
         std::unordered_map<RegexNFAByteState const*, uint32_t> const& state_ids
-) const -> std::string {
+) const -> std::optional<std::string> {
     std::vector<std::string> byte_transitions;
-    for (uint32_t idx = 0; idx < cSizeOfByte; idx++) {
+    for (uint32_t idx{0}; idx < cSizeOfByte; ++idx) {
         for (auto const* dest_state : m_bytes_transitions[idx]) {
-            byte_transitions.push_back(
+            byte_transitions.emplace_back(
                     fmt::format("{}-->{}", static_cast<char>(idx), state_ids.at(dest_state))
             );
         }
     }
+
     std::vector<std::string> epsilon_transitions;
     for (auto const* dest_state : m_epsilon_transitions) {
-        epsilon_transitions.push_back(std::to_string(state_ids.at(dest_state)));
+        epsilon_transitions.emplace_back(std::to_string(state_ids.at(dest_state)));
     }
+
     std::vector<std::string> positive_tagged_transitions;
     for (auto const& positive_tagged_transition : m_positive_tagged_transitions) {
-        positive_tagged_transitions.push_back(positive_tagged_transition.serialize(state_ids));
+        auto const serialized_positive_transition_it
+                = positive_tagged_transition.serialize(state_ids);
+        if (serialized_positive_transition_it.has_value()) {
+            positive_tagged_transitions.emplace_back(serialized_positive_transition_it.value());
+        } else {
+            return std::nullopt;
+        }
```

Comment:
```suggestion
        if (false == serialized_positive_transition_it.has_value()) {
            return std::nullopt;
        }
        positive_tagged_transitions.emplace_back(serialized_positive_transition_it.value());
```
I would probably write in this way to reduce the number of inner blocks

Reply:
Changed for both positive tags and negative tags.